src/modules/dolt_branch.rs

```diff
@@ -0,0 +1,186 @@
+use unicode_segmentation::UnicodeSegmentation;
+
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::dolt_branch::DoltBranchConfig;
+use crate::formatter::StringFormatter;
+use crate::utils::get_command_string_output;
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("dolt_branch");
+    let config: DoltBranchConfig = DoltBranchConfig::try_load(module.config);
+
+    let is_dolt_repo = context
+        .try_begin_scan()?
+        .set_files(&config.detect_file)
+        .set_folders(&config.detect_folder)
+        .is_match();
+    if !is_dolt_repo {
+        return None;
+    }
+
+    if config.disabled {
+        return None;
+    }
+
+    let len = if config.truncation_length <= 0 {
+        log::warn!(
+            "\"truncation_length\" should be a positive value, got {}",
+            config.truncation_length
+        );
+        std::usize::MAX
+    } else {
+        config.truncation_length as usize
+    };
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|variable, _| match variable {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "branch" => {
+                    let command = context.exec_cmd("dolt", &["branch", "--show-current"])?;
```

Comment:
This commands seems to be very slow (~300ms). Is there a way to speed this up? This module also doesn't seem to work on Windows right now, but that could be due to #3582.